rust/datafusion/src/execution/context.rs

```diff
@@ -383,24 +383,56 @@ impl ScalarFunctionRegistry for ExecutionContext {
     }
 }
 
+/// A planner used to add extensions to DataFusion logical and phusical plans.
```

Comment:
FYI @andygrove  here is the single trait you suggested (though I renamed it `QueryPlanner` rather than `UserDefinedQueryPlanner`